tests/kafkatest/services/kafka/kafka.py

```diff
@@ -91,12 +116,11 @@ def stop_node(self, node, clean_shutdown=True):
 
         for pid in pids:
             node.account.signal(pid, sig, allow_fail=False)
-
-        node.account.ssh("rm -f /mnt/kafka.pid", allow_fail=False)
+        wait_until(lambda: len(self.pids(node)) == 0, timeout_sec=20, err_msg="Kafka node failed to stop")
 
     def clean_node(self, node):
         node.account.kill_process("kafka", clean_shutdown=False, allow_fail=True)
-        node.account.ssh("rm -rf /mnt/kafka-logs /mnt/kafka.properties /mnt/kafka.log /mnt/kafka.pid", allow_fail=False)
+        node.account.ssh("rm -rf /mnt/kafka-logs /mnt/kafka.properties /mnt/kafka.log", allow_fail=False)
```

Comment:
Yeah, but you can't just consider normal operation. This `clean_node` has a `kill_process` for a reason -- it may be our last chance to kill a process that didn't shut down correctly during the test. If you nuke a pid file that doesn't belong to you, then some other process may not get cleaned up properly (I think this is probably one of the issues you were referring to in your previous comment about pid files being brittle, but other processes also shouldn't go around deleting stuff that isn't theirs :) ).

Anyway, this isn't critical here, but I think elsewhere we've moved everything for a `Service` under a dedicated directory in `/mnt`. Maybe the same approach should be used here so we could `rm -rf /mnt/kafka/` instead of all of `/mnt`?